### DISCUSSION TITLE
AWS Lambda (serverless) Question

### DISCUSSION THREAD

**Discussion by ogencoglu** [sybrenjansen/mpire#135] [General] [answered]
It is well known that AWS Lambda has problems with multiprocessing:
- https://stackoverflow.com/questions/61737183/python-multiprocessing-issue-with-aws-lambda-and-exchangelib
- https://stackoverflow.com/questions/34005930/multiprocessing-semlock-is-not-implemented-when-running-on-aws-lambda
- https://stackoverflow.com/questions/61737183/python-multiprocessing-issue-with-aws-lambda-and-exchangelib
- https://aws.amazon.com/blogs/compute/parallel-processing-in-python-with-aws-lambda/

Due to the Lambda execution environment not having `/dev/shm` (shared memory for processes) support, you can’t use multiprocessing.Queue or multiprocessing.Pool. You can use multiprocessing.Pipe though.

There are some solutions claimed: https://pypi.org/project/lambda-multiprocessing/

My questions is would mpire face the same issue in AWS Lambda?

**Comment by sybrenjansen** [comment, marked as answer]
Yes, mpire would face the same issue. It relies quite heavily on queues and semaphores/semlocks (which are used by e.g. lock/event/condition objects). 

Removing the reliance on those would be unfeasible, unfortunately
